src/nwbinspector/register_checks.py

```diff
@@ -120,20 +120,20 @@ def register_check_and_auto_parse(check_function):
 
         @wraps(check_function)
         def auto_parse_some_output(*args, **kwargs) -> InspectorMessage:
-
             if args:
                 obj = args[0]
             else:
                 obj = kwargs[list(kwargs)[0]]
             output = check_function(*args, **kwargs)
-            if isinstance(output, Iterable):
+            auto_parsed_result = None
+            if isinstance(output, InspectorMessage):
+                auto_parsed_result = auto_parse(check_function=check_function, obj=obj, result=output)
+            elif output is not None:
```

Comment:
Now this is an interesting thing I just discovered. With both `typing.Iterable` and `collections.abc.Iterable`, the claim `isinstance(list, Iterable)` fails. Not a huge issue since the `output` here is always a raw generator anyway (which appears to work fine) but it sends the wrong impression that we are truly checking if the returned value can be iterated, which we are apparently not.

I played and read around with detecting iterability and the best solution appear to really just have a helper function like

```python
def is_iterable(obj):
    try:
        iter(obj)
        return True
    except TypeError:
        return False
```

though it's likely that in most of our cases, a less obtrusive `hasattr(obj, "__iter__")` should also suffice...

Anyway that all got a bit messy so I decided the simplest case is actually to reverse the order of logic.

Test first if the output is an `InspectorMessage`, if not then check if it's `None`, if not then you know it has to be the iterable of messages.